doc/fileformat/serdes.rst

```diff
@@ -0,0 +1,262 @@
+.. _formatserdes:
+
+Checkpointing
+=============
+
+While not a fileformat on its own, checkpoints allow users to dump a snapshot of
+the simulation state for a later time. This is **not** intended for data
+extraction (see :ref:`probesample` for this functionality), but instead a reset
+the simulation state to this point in time. The saved data is not a complete
+image of the state, but strives to be a minimal subset from which to restore
+(see :ref:`below <impl guide>`).
+
+Checkpoints are currently _not_ portable; i.e. the following constraints apply:
+
+1. The recipe class must be the same across ``serialize`` / ``deserialize``
+   Unfortunately, we cannot enforce this at runtime, since we lack the
+   appropriate introspection capabilities in C++/Python.
+2. Likewise, ``context`` and ``domain_decomposition`` must be identical. Again
+   we cannot validate this, since we'd either stop migration between ranks or
+   would have to instantiate and sort every local domain decomposition across
+   all ranks.
+3. The hardware configuration must be identical: SIMD, GPU, and Scalar CPU code
+   have different binary layouts which makes serializing across hardware settings
+   invalid.
+
+Thus, the recommended approach is to treat a simulation 'script' (``.cxx`` /
+``.py``), the parallel-runtime parameters (think ``mpirun``) and the associated
+snapshots as a single unit.
+
+.. note:: While this functionality can quite obviously be abused to mutate certain
+          variable out-of-band, we strongly advise against it.
+
+Usage
+-----
+
+Checkpoints are independent of the storage engine used, it allows for multiple
+backends. The general usage is show in the examples below; we assume familiarity
+with the C++ and Python recipe interfaces:
+
+  .. code:: c++
```

Comment:
The extra indent before `.. code::` creates a blockquote. Dedent pls. Same for all following code blocks.